frontend/src/metabase/css/components/modal.css

```diff
@@ -10,6 +10,10 @@
 .Modal.Modal--medium  { width: 65%; }
 .Modal.Modal--wide    { width: 85%; }
 
+.Modal.Modal--tall {
```

Comment:
If the size is based purely on the size of the content of the modal it will jump around as you toggle between sections.